### PR TITLE
Loosen `.` file detection, add `--extra-ignore-contents`

### LINKED ISSUE #624: jupyterlite build fails if --contents dir has any dir with a `.` in name
## Description

Running `jupyter lite build --contents contents` fails with the following error if there is any directory whose name starts with a `.`.

```
❯ jupyter lite build --contents contents                                                                             130 ↵
static:jupyter-lite.json
.  pre_status:static:jupyter-lite.json
    tarball:         jupyterlite-app-0.1.0-beta.5.tgz 8MB
    output:          /Users/yuvipanda/code/jupyterlite-data8/tmp/_output
    lite dir:        /Users/yuvipanda/code/jupyterlite-data8/tmp
    apps:            
    sourcemaps:      True
    unused packages: True
archive:archive
contents:contents
lite:jupyter-lite.json
mathjax:status
pyodide:pyodide
serve:contents
settings:overrides
translation:translation
.  status:archive:archive
[LiteBuildApp] No archive (yet): tmp-jupyterlite.tgz
.  status:contents:contents
    contents: 1 files
.  status:lite:jupyter-lite.json
[LiteBuildApp]     jupyter-lite.(json|ipynb): 0 files
.  status:mathjax:status
     jupyter-server-mathjax: /Users/yuvipanda/.local/share/virtualenvs/textbook/lib/python3.9/site-packages/jupyter_server_mathjax/static 
     configured MathJax dir: /Users/yuvipanda/code/jupyterlite-data8/tmp/traitlets.Undefined
     effective MathJax dir: /Users/yuvipanda/.local/share/virtualenvs/textbook/lib/python3.9/site-packages/jupyter_server_mathjax/static
.  status:pyodide:pyodide
     URL: None
 archive: []
   cache: 0 files
   local: 0 files
.  status:serve:contents
    url: http://127.0.0.1:8000/
    server: tornado
    headers:
.  status:settings:overrides
    overrides.json: 0
.  status:translation:translation
    translation files: 0 files
static:output_dir
.  pre_init:static:output_dir
static:unpack
.  init:static:unpack
federated_extensions:copy:ext:jupyterlab_pygments
.  pre_build:federated_extensions:copy:ext:jupyterlab_pygments
contents:copy:.binder/wat
mathjax:copy
translation:copy
.  build:contents:copy:.binder/wat
.  build:mathjax:copy
.  build:translation:copy
contents:contents:.binder
contents:contents:.
federated_extensions:patch
mathjax:patch
.  post_build:contents:contents:.binder
TaskError - taskid:post_build:contents:contents:.binder
PythonAction Error
Traceback (most recent call last):
  File "/Users/yuvipanda/.local/share/virtualenvs/textbook/lib/python3.9/site-packages/doit/action.py", line 461, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/Users/yuvipanda/.local/share/virtualenvs/textbook/lib/python3.9/site-packages/jupyterlite/addons/contents.py", line 170, in one_contents_path
    listing = fm.get(listing_path)
  File "/Users/yuvipanda/.local/share/virtualenvs/textbook/lib/python3.9/site-packages/jupyter_server/services/contents/filemanager.py", line 370, in get
    raise web.HTTPError(404, "No such file or directory: %s" % path)
tornado.web.HTTPError: HTTP 404: Not Found (No such file or directory: binder)
```

## Reproduce

Create a directory named `.binder`, and just put any random file inside it. 

```
mkdir -p contents/.binder
touch contents/.binder/runtime.txt
jupyter lite build --contents contents
```

## Context
Output from `pip list`:

```
❯ pip list                                                                                                           130 ↵
Package            Version
------------------ -------
cloudpickle        2.0.0
doit               0.36.0
entrypoints        0.4
importlib-metadata 4.11.3
jupyter-core       4.10.0
jupyterlite        0.1.0b5
pip                22.0.4
setuptools         60.10.0
traitlets          5.1.1
zipp               3.8.0
```

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #624 

## Code changes

- [x] fix stripping of leading `.` (now strips the full `files` path)
- [x] add `--extra-ignore-contents` to augment the base `--ignore-contents` without replacing
- [x] complain in addon with some help docs referencing above, or `allow_hidden`
- [x] add test
- [x] add hidden files section in docs

## User-facing changes

- more repos with `.binder` and other related folders would build as-is
  - it remains to be seen if these would actually be _serverd_ by many http servers

## Backwards-incompatible changes

- n/a, as it just failed hard before

### CODE REVIEW

**Comment by bollwyvl** [comment]
It looks like hidden files make it all the way to the browser. With an `overrides.json` like:

```
{"@jupyterlab/filebrowser-extension:browser": {"showHiddenFiles": true}}
```

...they appear greyed out, as expected:

Screenshot from 2022-11-01 19-15-59

and can be updated, saved, access from inside pyolite, etc.

**Comment by yuvipanda** [comment]
@bollwyvl ok, I managed to test this! With this PR applied, the default build no longer breaks (yay!). Thank you!

**Comment by bollwyvl** [comment]
:bellhop_bell:  @jtpio @martinRenou

File: py/jupyterlite/src/jupyterlite/config.py [hunk 1]
```diff
@@ -115,6 +115,10 @@ class LiteBuildConfig(LoggingConfigurable):
         help="Path regular expressions that should never be included as contents"
     ).tag(config=True)
 
+    extra_ignore_contents: _Tuple[_Text] = Tuple(
```
[the remark this answers is not in the document]
Comment (bollwyvl) [reply py/jupyterlite/src/jupyterlite/config.py:118]: yeah, dunno... having no way to get rid of defaults is kinda brutal, which is why that was there in the first place...

[the remark this answers is not in the document]
Comment (jtpio) [reply py/jupyterlite/src/jupyterlite/config.py:118]: > get rid of defaults

ah right, then it makes sense to keep it :+1:

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
FYI @yuvipanda this is now available in `0.1.0b15`: https://github.com/jupyterlite/jupyterlite/releases/tag/v0.1.0b15

**Pull request merged**